azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/AzureSpringCloud.java

```diff
@@ -50,38 +53,56 @@ private AzureSpringCloud(@Nonnull final List<Subscription> subscriptions) { // f
         super(AzureSpringCloud::new, subscriptions);
     }
 
-    @Nonnull
-    public SpringCloudCluster cluster(@Nonnull SpringCloudClusterEntity cluster) {
-        final AppPlatformManager client = this.getClient(cluster.getSubscriptionId());
-        return new SpringCloudCluster(cluster, client);
-    }
-
     @Nullable
     public SpringCloudCluster cluster(@Nonnull String name) {
         return this.clusters().stream()
-            .filter((s) -> Objects.equals(s.name(), name))
-            .findAny().orElse(null);
+                .filter((s) -> Objects.equals(s.name(), name))
+                .findAny().orElse(null);
+    }
+
+    @Nonnull
+    SpringCloudCluster cluster(@Nonnull SpringService remote) {
+        return this.cluster(new SpringCloudClusterEntity(remote));
+    }
+
+    @Nonnull
+    private SpringCloudCluster cluster(@Nonnull SpringCloudClusterEntity cluster) {
+        return new SpringCloudCluster(cluster, this.getClient(cluster.getSubscriptionId()));
     }
 
     @Nonnull
     public List<SpringCloudCluster> clusters() {
         return this.getSubscriptions().stream()
-            .map(s -> getClient(s.getId()))
-            .map(SpringCloudClusterManager::new)
-            .flatMap(m -> m.getAll().stream())
-            .map(this::cluster)
-            .collect(Collectors.toList());
+                .map(s -> getClient(s.getId()))
+                .flatMap(c -> c.list().stream())
+                .map(this::cluster)
+                .collect(Collectors.toList());
+    }
+
+    @Cacheable(cacheName = "SpringServices", key = "$subscriptionId")
+    protected SpringServices getClient(final String subscriptionId) {
+        return getAzureResourceManager(subscriptionId).springServices();
     }
 
-    @Cacheable(cacheName = "AppPlatformManager", key = "$subscriptionId")
-    protected AppPlatformManager getClient(final String subscriptionId) {
+    @Cacheable(cacheName = "AzureResourceManager", key = "$subscriptionId")
+    private AzureResourceManager getAzureResourceManager(String subscriptionId) {
```

Comment:
we shall replace AzureResourceManager with  dedicated manager

Reply:
yes, I will fix it.